lib/src/main/kotlin/com/swmansion/starknet/data/serializers/TransactionPayloadSerializer.kt

```diff
@@ -65,3 +67,58 @@ object DeclareTransactionV2PayloadSerializer : KSerializer<DeclareTransactionV2P
         throw SerializationException("Class used for serialization only.")
     }
 }
+
+@OptIn(kotlinx.serialization.ExperimentalSerializationApi::class)
+object InvokeTransactionPayloadSerializer : KSerializer<InvokeTransactionPayload> {
+
+    override val descriptor: SerialDescriptor
+        get() = DeclareTransactionV1Payload.serializer().descriptor
+
+    override fun serialize(encoder: Encoder, value: InvokeTransactionPayload) {
+        require(encoder is JsonEncoder)
+
+        val jsonObject = buildJsonObject {
+            put("sender_address", value.senderAddress.hexString())
+            putJsonArray("calldata") { value.calldata.forEach { add(it) } }
+            putJsonArray("signature") { value.signature.forEach { add(it) } }
+            put("max_fee", value.maxFee.hexString())
+            put("version", value.version)
+            put("nonce", value.nonce)
+            put("type", value.type.toString())
+        }
+
+        encoder.encodeJsonElement(jsonObject)
+    }
+
+    override fun deserialize(decoder: Decoder): InvokeTransactionPayload {
+        throw SerializationException("Class used for serialization only.")
+    }
+}
+
+@OptIn(kotlinx.serialization.ExperimentalSerializationApi::class)
+object DeployAccountTransactionPayloadSerializer : KSerializer<DeployAccountTransactionPayload> {
+
+    override val descriptor: SerialDescriptor
+        get() = DeclareTransactionV1Payload.serializer().descriptor
+
+    override fun serialize(encoder: Encoder, value: DeployAccountTransactionPayload) {
+        require(encoder is JsonEncoder)
+
+        val jsonObject = buildJsonObject {
+            put("class_hash", value.classHash.hexString())
+            put("contract_address_salt", value.salt.hexString())
+            putJsonArray("constructor_calldata") { value.constructorCalldata.forEach { add(it) } }
+            put("version", value.version)
+            put("nonce", value.nonce)
+            put("max_fee", value.maxFee.hexString())
+            putJsonArray("signature") { value.signature.forEach { add(it) } }
+            put("type", value.type.toString())
+        }
+
+        encoder.encodeJsonElement(jsonObject)
+    }
+
+    override fun deserialize(decoder: Decoder): DeployAccountTransactionPayload {
+        throw SerializationException("Class used for serialization only.")
+    }
+}
```

Comment:
why those are needed? I though the data classes were already `@Serializable` and not much transformation is needed?

Reply:
They are, but the fail to serialise `type` field when `encodeToJsonElement` is used. 
Since we already have `DeclareTransactionV1PayloadSerializer` and `DeclareTransactionV2PayloadSerializer` I think it would be best way to solve this.